static/src/javascripts/bootstraps/enhanced/liveblog.js

```diff
@@ -80,6 +81,30 @@ const componentEvent = (pinnedBlockId, action, value) => ({
     }
 });
 
+const trackPinnedPostDuration = (pinnedBlock) => {
+    let hasBeenSeen = false;
+    const pinnedPostTiming = measureTiming('pinned-post-view-duration');
+    const originalPinnedBlockId = pinnedBlock.dataset.blockId
+    const onIntersect = (entries) => {
+        entries
+            .forEach((entry) => {
+                if (entry.isIntersecting) {
+                    hasBeenSeen = true;
+                    pinnedPostTiming.start();
+                } else if (hasBeenSeen) {
+                    const timeTaken = pinnedPostTiming.end();
+                    if (timeTaken) {
+                        ophan.record(componentEvent(originalPinnedBlockId, 'DURATION', {value: timeTaken}));
+                    }
+                }
+            });
+    };
+    const observer = new IntersectionObserver(onIntersect, {
```

Comment:
Nice, I added the condition `if ('IntersectionObserver' in window)` but wasn't sure if I should bring a fallback. Do you think we'd better having a fallback?

Reply:
I think it's fair to fallback to not doing anything. In some situations there are other approaches you can fallback to but I think in this case, where the duration is important, it might not be possible